tiledbimg/converters/ome_zarr.py

```diff
@@ -1,24 +1,134 @@
+            return temp_array.reshape(tczyx_shape)
+
+    def level_metadata(self, level: int) -> Dict[str, Any]:
+        with tiledb.open(self._levels[level]) as L:
+            return dict(pickle.loads(L.meta["pickled_zarrwriter_kwargs"]))
+
+    def metadata(self) -> Dict[str, Any]:
+        return dict(pickle.loads(self.tldb_image.meta["pickled_zarrwriter_kwargs"]))
 
-from .base import ImageConverter, ImageReader
+    def write(
+        self,
+        images: Sequence[np.ndarray],
+        level_metas: Sequence[Dict[str, Any]],
+        image_meta: Dict[str, Any] = {},
+    ) -> None:
+        level_metas_zarray = []
+        for level_meta in level_metas:
+            zarray_meta = level_meta.get("zarray")
+            if zarray_meta is not None:
+                del zarray_meta.get("compressor")["id"]
+                zarray_meta["compressor"] = Blosc.from_config(
+                    zarray_meta.get("compressor")
+                )
+            level_metas_zarray.append(zarray_meta)
+
+        # Write image does not support incremental pyramid write
+
+        write_multiscale(
+            images,
+            self.image_group,
+            scaler=None,
+            axes=["t", "c", "z", "y", "x"],
+            storage_options=level_metas_zarray,
+            metadata=image_meta.get("metadata"),
+        )
 
 
 class OMEZarrReader(ImageReader):
     def __init__(self, input_path: str):
-        self._zarray = zarr.open(input_path)
+        self.image = Reader(ZarrLocation(input_path))
+        resolutions = [
+            img_path for img_path in glob.glob(f"{input_path}/[!OME]*", recursive=True)
+        ]
+        self.res_nodes = [
+            nodes for res in resolutions for nodes in Reader(ZarrLocation(res))()
+        ]
+        self._levels = []
+        for layer in self.res_nodes:
+            self._levels.append(Level(layer))
 
     @property
     def level_count(self) -> int:
-        return len(self._zarray)
+        return len(self._levels)
 
     def level_image(self, level: int) -> np.ndarray:
-        zarray_l0 = self._zarray[level][0]
-        return np.asarray(zarray_l0).squeeze()
+        leveled_zarray = self._levels[level].node.data[0]
+        # From NGFF format spec there is guarantee that axes are t,c,z,y,x
+        return np.asarray(leveled_zarray).squeeze()
+
+    def level_metadata(self, level: int) -> Dict[str, Any]:
+        meta_root = self._levels[level].node.zarr
+        writer_kwargs = dict(
+            fmt=meta_root.fmt,
+            root_attrs=meta_root.root_attrs,
+            zarray=meta_root.zarray,
+            zgroup=meta_root.zgroup,
+        )
+        return {"pickled_zarrwriter_kwargs": pickle.dumps(writer_kwargs)}
+
+    def metadata(self) -> Dict[str, Any]:
+        writer_kwargs = dict(
+            bioformats2raw_layout=3,
+            zarr_format=self.image.zarr.zgroup.get("zarr_format", 0),
+            metadata=self.image.zarr.root_attrs.get("multiscales")[0].get("metadata"),
```

Comment:
1. This will throw if there is no `multiscales` key or it is empty
2. Why save only the first `multiscales` element? If it's expected to have only one, an assertion should be added.